### PR TITLE
chore(deps): upgrade redis

### PR DESCRIPTION
**Pull request by traviscrist**
## Changes proposed in this pull request

- Upgrades Redis to 5 from 4. 
- Types are added, see https://github.com/luin/ioredis/wiki/Upgrading-from-v4-to-v5 for details

## Checklist

- [ ] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass

### CODE REVIEW

File: packages/backend/src/connector/index.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import IORedis from 'ioredis'
+import Redis from 'ioredis'
```
Comment (wilsonianb) [inline packages/backend/src/connector/index.ts:1]: There's a few other `import IORedis from 'ioredis'` if we want to change them all to be consistent
https://github.com/interledger/rafiki/search?q=IORedis

**Review by wilsonianb — COMMENTED** [review]
merge `main`

File: packages/backend/src/connector/index.ts (hunk 1 above)
Comment (traviscrist) [reply packages/backend/src/connector/index.ts:1]: Nice catch!

**Comment by wilsonianb** [comment]
`Build / Release Backend Docker Image / build-and-push` timed out?

**Review by wilsonianb — APPROVED** [review]

**Pull request merged**